### ISSUE TITLE
Feature Request List

### ISSUE THREAD

**Issue by Starfire13** [dcs-retribution/dcs-retribution#285]
### Is your feature request related to a problem? Please describe.

As requested, I am compiling a list of the various features I'd like to see in Retribution.

- [x] **1. Make CAS mission plannable over control points.**

With air assaults now being such a good way to capture enemy airfields and FOBs, it would be nice to have some way to provide CAS for the troops that are dropped off. I've been planning such missions over the frontline and manually dragging the waypoints over the control point for a while now, and while that works fine, this isn't an option on the campaigns that have no frontline.

I'm finding control point CAS to be very handy because more often than not, there's some pesky single half dead ground unit somewhere that the BAI flights didn't manage to demolish, preventing capture. Having a CAS flight swing by and clear out those few remaining ground units for the air assault troops is super helpful.

- [ ] **2. Definable campaign-specific win/loss conditions**

It would be good if we had win/loss conditions that extended beyond just capturing every control point. I think this would help diversify the types of campaign scenarios that are possible. This could be defined in the campaign yaml. As an example, the ability to win a campaign by destroying a specific map object strike target. This could be something like a vital bridge heavily defended by AAA.

- [ ] **3. Campaign win pop up with stats**

When a campaign is won, it would be good if we got a pop up showing the count for all the various units killed by type. Kinda like the end of turn screen, but across the entire campaign. And with stats for both OPFOR and OWNFOR. We could even include the air and ground unit count graphs that are already available in Retribution.

- [x] **4. Campaign options settings autoload from Default.zip**

At the moment, we have to click "load settings", select the campaign options settings zip file, and click open each time we start a campaign. It would be nice if we could create a default settings file (e.g. Default.zip) and that just autoloads as the default. I pretty much always use the same settings.

- [x] **5. Have some way to set "Squadrons start at full capacity" to default on**

I'd like some way to have this set to default on so I don't have to keep remembering to click it each time. Would be good to move it under the Gameplay section of the Mission Generator settings so you can load it from the Campaign Options settings file.

- [ ] **6. Plannable DEAD mission against static armour groups that have a secondary air defence group**

Some static armour ground unit groups have extra tough air defence (e.g. SA-15 TOR). But as it is, a BAI flight will always attack the primary group first and ignore the secondary group until the primary group is dead. This forces us to use longer range stand-off BAI loadouts for the AI so they don't all get shot down. SEAD Sweep is one way to reduce the impact of this problem, but it is not a perfect solution because it is a search and destroy in area task rather than a targeted one. It would be good if there were a way to plan a DEAD mission that targets the secondary group directly for static armour groups, so we could clear out the group's air defence before bombing them.

- [ ] **7. More intelligent grouping of frontline units**

I realise the frontline ground units have the intelligence of a baked potato, and perhaps trying to keep them alive is a lost cause. However, maybe we can help them die in a prettier manner. I'd like to suggest that rather than deploying ground units by order of purchase, to pick a selection from the pool and to then deploy them in mixed groups. This way, if you buy 10 Abrams and 10 Linebackers, you won't always have 10 Abrams on and 5 Linebackers on the 15-unit frontline, and 5 Linebackers in reserve. What's more, the Linebackers won't all be clustered to one side of the frontline. Rather, a mix of 15 Abrams and M113s will be spread evenly across the frontline. This is particularly useful for air defence units, so that they're actually embedded with different armour groups along the frontline so they can defend other units against CA flights instead of being all clustered together.

- [x] **8. Loadout selection drop down menu in Create Flight**

At the moment, changing the loadout for a flight requires creating the flight, then entering the Edit Flight pane, selecting the Payload tab, and changing it in there. What would be a nice QOL feature is if we could select the loadout for a flight right on the Create Flight pane. This is very handy because I often have to switch to alternate loadout presets for AI flights. For example, I want some SEAD flights to carry HARMs, others to carry TALDs. Or I might want certain BAI flights to use mavericks instead of cluster bombs if they are attacking static armour group that have SHORAD.

- [ ] **9. Show runway destruction in mid-mission progress tracker**

At the moment, the mid-mission progress tracker tracks unit deaths, base captures, etc. But it doesn't track runway destruction. Would be nice if runway destruction also shows up there.

- [ ] **10. Air Wing Configuration faction tabs should be renamed to OWNFOR and OPFOR**

At the moment, they are labelled "Blue" and "Red" which would not always be accurate since you can play as Redfor in Retribution. Heck, you can even play Red vs Red if you wanted to, or Blue vs Green.

### Describe the solution you'd like

Solution I'd like is for Stoner to win a billion dollars in the lottery so he can work on Retribution full time.

**Comment by Starfire13** [comment]
- [ ] 11. F-15Es still need the #T in front of additional target waypoints. This has to be done manually for every single mission right now and it's a hassle.

**Comment by Starfire13** [comment]
- [ ] 12. Join waypoints need to be moved further back as they are too far from the hold waypoint right now and very near to the target. Escort flights should be joining up with their escortees much earlier, not when they're a minute away from their target.

**Comment by Starfire13** [comment]
13. Number of bombing tasks for strike missions seems to be capped at 4, but there should be as many as there are targets at that strike target, like it is implemented in Liberation.

**Comment by Starfire13** [comment]
- [x] 13. Strike missions are currently planned without Group Attack checked, resulting in just the first aircraft in the group dropping bombs, and the others doing nothing.

**Comment by Raffson** [comment]
Regarding 13, wouldn't it be better to have group-attack only be checked if it's a bomber? And revert back to the "drop half of your payload" approach for static targets when guided bombs are being used with non-bomber aircraft?

**Comment by Starfire13** [comment]
I don't think so, no. I can't imagine a scenario where you'd want to just throw away half your ordnance at once with guided bombs. The whole point of guided bombs is the added precision and efficiency. One bomb to kill each target. I'd prefer each one hitting its own target rather than dropping a whole bunch on the same one.

Besides, it's a waste if you have 4 aircraft flying out there and three of the aircraft are just there for moral support and do nothing.

Edit: Stoner's new implementation is actually better than using group attack. Problem solved!

**Comment by Starfire13** [comment]
- [x] 14. It might be good at some point to look into how to add the Currenthill mods without ending up with a massive expansion of the current checklist of mods just because there are so many separate units which are all separate downloads. One possibility is to just enable Currenthill mods with a single checkbox, and then give it its own dedicated checklist, like is done with plugin options.

Edit: Looks like  Currenthill have now grouped his mods into packs for each nation rather than by single units. This should make things significantly easier as there are only half a dozen mod packs now rather than something like a hundred separate unit mods.

**Comment by Starfire13** [comment]
- [x] 15. Allow the option for selected control points to start off neutral with all air defence and static armour group positions empty (perhaps by ticking the "hide not flyable" checkbox?). This would also mean that the enemy should plan air assault missions against such airfields, so it becomes a contest to see who captures the control point first.

**Comment by Starfire13** [comment]
- [ ] 16. Allow selected air defence or static armour groups at friendly control points to start off as destroyed at campaign start. One could also have squadrons starting at half strength, with static aircraft flagged as dead so they show up as wrecks in the parking spaces at the airfield. This would allow the creation of scenarios that begin shortly after an enemy attack has severely crippled your air force and air defences, meaning you have to choose between beefing up your tattered air defence or replacing your destroyed aircraft on the following turns.

**Comment by Starfire13** [comment]
- [x] 17. This is something that should preferably be implemented either at the same time as or after #103, but it would be nice if certain mission types CAS and OCA/Aircraft that require visual ID of targets have their altitude from ingress split set to under the cloud layer for certain weather presets like all the ones with overcast clouds. The reason is the AI will simply RTB if it doesn't have visual contact with targets. I've gotten around this issue in the past by either manually changing waypoint altitudes in the mission editor or changing it in-mission with combined arms, but it would be nice if this could be automated.
